types/app.d.ts

```diff
@@ -1,5 +1,5 @@
 declare namespace tinyapp {
-  type Query = Record<string, string | number>;
+  type Query = Record<string, string | number | undefined>;
```

Comment:
`Query` is deserialized from 'URL', so I think `undefined` is not the possible value.

Reply:
We have the case where we use context to access to a page, which depends on the context and then we need 1 extra query param.

example:
```
interface PageChatQuery extends tinyapp.Query {
  context: 'agent' | 'driver';
  bookingId?: string;
}
```

In this case, we can deserialize the query, but we need to control the undefined in our code. Do you have any suggestion for this case without use undefined?

Thanks!!!! :)

Reply:
Ok, we solve it with conditional types, not bad though. Check the code, it is updated!